File: class2relational/src/main/java/org/atlanmod/model/dynamic/relationalModel/ColumnToTable.java
package org.atlanmod.model.dynamic.relationalModel;

import org.atlanmod.model.dynamic.DynamicElement;

import java.util.ArrayList;
import java.util.List;

public class ColumnToTable extends RelationalLink {

    public ColumnToTable(RelationalColumn source, RelationalTable target) {
        super(RelationalModel.COLUMN_TABLE, source, null);
        List<DynamicElement> new_target = new ArrayList<>();
        new_target.add(target);
        this.setTarget(new_target);
    }

    public String toString() {
        return "(" +
                ((RelationalElement) this.getSource()).getId() +
                ", " + super.getType() +
                ", " + ((RelationalElement) this.getTarget().get(0)).getId() +
                ")";
    }
}

File: relational2class/src/main/scala/org/atlanmod/model/relationalmodel/ColumnToTable.scala
package org.atlanmod.model.relationalmodel

class ColumnToTable(source: RelationalColumn, target: RelationalTable)
  extends RelationalLink(RelationalMetamodel.COLUMN_TABLE, source, List(target)){

    override def toString: String =
        "(" + source.getId() + ", " + super.getType + ", " + target.getId() + ")"

    override def getSource: RelationalColumn = source

    override def getTarget: List[RelationalTable] =  List(target)

    def getTargetTable : RelationalTable = target

}

